### ISSUE TITLE
[HOLD for payment 2023-08-28] [$1000] IOU - LAK currency is not displayed correctly inside the badge

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#23235]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open Expensify stg app
2. Click the FAB button
3. Click on "Split bill"
4. Click on the currency
5. Select "LAK-LAK" from the list of currencies
6. Enter any 8 digit number
7. Click on the "Next" button
8. Selects any 2 contacts
9. Press the "Next" button
10. Click on the "Split {Currency}{Amount}" button

## Expected Result:
The amount in LAK currency on the badge is displayed correctly and is equal to the amount owed by the participant

## Actual Result:
The amount in LAK currency on the badge is displayed incorrectly, the last two digits of the amount owed are missing

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.43.0

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/57eca6a8-6448-400b-bc12-b9fd547ccf85

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0110273979413679a9</li>
        <li>Upwork Job ID: 1683641607558184960</li>
        <li>Last Price Increase: 2023-08-01</li>
    </ul>
</details>

**Comment by BhuvaneshPatil** [comment]
The problem needs to be solved from backend.
This problem is with another currencies as well.
Ex - LBP.

The property coming from backend, ie. `lastMessageText` and `lastMessageHTML` is calculated wrong

**Comment by sophiepintoraetz** [comment]
<img width="1505" alt="image" src="https://github.com/Expensify/App/assets/42940078/5f4fcc26-0a1a-4594-9da7-59acf4a25c36">

Assigning @sobitneupane to confirm that this is a backend issue!

**Comment by sobitneupane** [comment]
@sophiepintoraetz It looks like a backend issue to me as well. It works fine in offline mode but as soon as connected to internet it changes to the wrong value. I believe it is due to some response from the backend.

https://github.com/Expensify/App/assets/25876548/bea8c2b3-48bf-413c-95ac-7009f9188c3e

**Comment by aldo-expensify** [comment]
I didn't have time to investigate this today, I'll try early next week.

**Comment by sophiepintoraetz** [comment]
@aldo-expensify - hopefully this week?

**Comment by aldo-expensify** [comment]
This is possibly because of:

> Some code was added [here](https://github.com/Expensify/Web-Expensify/pull/37942/files) changing the formatting used for IOU in the backend, but we didn't do anything in the App code to be consistent. This also makes more spread the code handling decimals in the backend. In the past this lived only in PHP-Libs https://github.com/Expensify/PHP-Libs/blob/main/src/CurrencyInfo.php.

I have already an issue for this: https://github.com/Expensify/Expensify/issues/300539

I'll try to give it time later during the week so we can get it done. Meanwhile, I think we can remove the help wanted and make it a weekly?

**Comment by aldo-expensify** [comment]
Removing `Help Wanted` for now as this is most likely a backend problem

**Comment by aldo-expensify** [comment]
I don't think this is happening anymore

https://github.com/Expensify/App/assets/87341702/8c9cebb3-f73d-4337-9b38-dee1e76c27dd

and with a bigger number `12345678`:

<img width="982" alt="image" src="https://github.com/Expensify/App/assets/87341702/eade7bc1-f50f-4786-9132-fde5196a73f6">

**Comment by aldo-expensify** [comment]
This feels like a duplicate of https://github.com/Expensify/App/issues/19792 and it was solved when the solution for that ones deployed I think.

**Comment by aldo-expensify** [comment]
Oops, I think I was having problems with my dev env and this is not working really! reopening to keep investigating

**Comment by aldo-expensify** [comment]
Confirmed that this bug still exists:

<img width="868" alt="image" src="https://github.com/Expensify/App/assets/87341702/41da698a-5aea-4d04-b2fb-0f0e88fe2fa7">

**Comment by aldo-expensify** [comment]
The frontend seems to be sending an incorrect amount to the backend. The amount should be in "cents" (1/100 subunit) independent of the currency used.

Creating a money request of 100 USD (2 decimals currency), should send an mount of 10,000 to the backend
Creating a money request of 100 LAK (0 decimals currency), should send an mount of 10,000 to the backend

**Comment by aldo-expensify** [comment]
Frontend PR up: https://github.com/Expensify/App/pull/24175

**Comment by aldo-expensify** [comment]
I wasn't able to finish, my PR is causing a regression described here: https://github.com/Expensify/App/pull/24175#issuecomment-1666333579

I'll continue on Tuesday 👋

**Comment by aldo-expensify** [comment]
Didn't continue on Tuesday, but I finished the code for https://github.com/Expensify/App/pull/24175 today. I just have to complete the testing steps and provide more screenshots, then I'll make it ready for review.

**Comment by aldo-expensify** [comment]
I made the PR ready for review today.

**Comment by sophiepintoraetz** [comment]
@sobitneupane - can you put in a request for $1000 once you have completed the checklist above?

**Comment by sophiepintoraetz** [comment]
Bump @sobitneupane

**Comment by sobitneupane** [comment]
Sorry for the delay. I will try to get to it by tomorrow EOD.

**Comment by aldo-expensify** [comment]
Bump @sobitneupane

**Comment by sophiepintoraetz** [comment]
Closing in the meantime - seeing as the action is with @sobitneupane and the request summary is [here](https://github.com/Expensify/App/issues/23235#issuecomment-1695016076).

**Comment by sobitneupane** [comment]
> **BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

I think this issue was caused by some backend change as suggested [here](https://github.com/Expensify/App/issues/23235#issuecomment-1659351483)

**Comment by sobitneupane** [comment]
**Regression Test Proposal:**

1. Create money requests in currencies that have decimals and currencies without decimals, for example:
    - BHD: 3 decimals, the third decimal will be truncated because we can't store it in the database, but it still should display with 3 decimal places after the . (the third decimal will always be a 0)
    - JPY: 0 decimals
    - VND: 0 decimals
    - USD: 2 decimals
    - RSD: 2 decimals (special case, should be tested)
 
2. Verify that preview looks correctly formatted in chat report and in LHN. 

Do we agree 👍 or 👎

**Comment by sobitneupane** [comment]
Requested payment on newDot.

https://github.com/Expensify/App/issues/23235#issuecomment-1695016076

**Comment by JmillsExpensify** [comment]
$1,000 approved for payment in NewDot based on this [BZ comment](https://github.com/Expensify/App/issues/23235#issuecomment-1695016076).
